fork-off/src/chainspec_combining.rs

```diff
@@ -48,7 +49,7 @@ pub fn combine_states(
                 removed_per_path_count.bump(path);
                 false
             }
-            None => true,
+            None => !well_known_keys::is_child_storage_key(&Into::<Vec<u8>>::into(k)), // remove child keys, they are not accepted as part of storage, see https://github.com/Cardinal-Cryptography/substrate/blob/98c2eeea74413044ae8ccfca1b6d56d01b57a76b/client/db/src/lib.rs#L778
```

Comment:
Rust newb question: why do we need `Into::<Vec<u8>>::into(k)`? I would have expected `k.into()` or maybe `k.into::<Vec<_>>()` to work.

Reply:
Substrate newb question: why is it OK to drop these keys during fork-off? Are they somehow unimportant?

Reply:
`Into::<Vec<u8>>::into(k)` should be equivalent to `k.into::<Vec<_>>()`; `k.into()` might be ambiguous though - depending on the context

Reply:
ad1) into() doesn't take generic arguments:
```
    fn into(self) -> U {
        U::from(self)
    }
```
ad2) see code-comment and the PR description

Reply:
> ad2) see code-comment and the PR description

I did read those, and I can see that they are not accepted. What I'm wondering is why is it OK to just remove some keys? Seems like that will cause the forked-off chain to have a different state in some way, and what I don't get is in what way exactly.

Reply:
:+1: